### PR TITLE
Update XGBoost 0.82

### PR DESCRIPTION
**Pull request by ViralBShah**

### CODE REVIEW

**Comment by staticfloat** [comment]
Try removing the part that copies `libgomp` over and instead expressing a dependency on `CompilerSupportLibraries_jll`.

**Comment by ViralBShah** [comment]
> Try removing the part that copies `libgomp` over and instead expressing a dependency on `CompilerSupportLibraries_jll`.

That was timely. Thanks!

**Comment by aviks** [comment]
So using this artifacts, tests fail on Mac (but pass on Linux and Windows. Failure on Mac looks like this: 

```
dyld: lazy symbol binding failed: Symbol not found: ___cxa_thread_atexit

  Referenced from: /Users/travis/.julia/artifacts/3ff38e086f3b040acfb66c0e7d42fac2bdc6eea5/lib/libxgboost.dylib

  Expected in: /Users/travis/.julia/artifacts/7c9ef733699a1d86b8a6073ed08a4457e3e790f7/lib/libstdc++.6.dylib

dyld: Symbol not found: ___cxa_thread_atexit

  Referenced from: /Users/travis/.julia/artifacts/3ff38e086f3b040acfb66c0e7d42fac2bdc6eea5/lib/libxgboost.dylib

  Expected in: /Users/travis/.julia/artifacts/7c9ef733699a1d86b8a6073ed08a4457e3e790f7/lib/libstdc++.6.dylib

signal (6): Abort trap: 6

in expression starting at /Users/travis/build/dmlc/XGBoost.jl/test/runtests.jl:8

signal (6): Abort trap: 6

in expression starting at /Users/travis/build/dmlc/XGBoost.jl/test/runtests.jl:8

unknown function (ip: 0x1098c5249)

Allocations: 10016393 (Pool: 10014671; Big: 1722); GC: 11
```

**Comment by giordano** [comment]
Better open an issue here in Yggdrasil, otherwise this comment will likely be lost

**Comment by ViralBShah** [comment]
Note that the xgboost darwin build has a ton of warnings and scary looking messages. Any idea what can fix these? Perhaps that will also fix the issue.

https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=2228&view=logs&j=0e6f23d3-bbc5-5a5c-1679-8a7f37dad18d&t=62f65e56-6499-50ca-7cf6-50d14fe18c3b&l=57457

File: X/XGBoost/build_tarballs.jl [hunk 1]
```diff
@@ -4,9 +4,8 @@ using BinaryBuilder, Pkg
 name = "XGBoost"
 version = v"0.82"
 sources = [
-    "https://github.com/dmlc/xgboost.git"=>
-    "3f83dcd50286d7c8d22e552942bd6572547c32b9",
-    "./bundled",
+    GitSource("https://github.com/dmlc/xgboost.git","bf3241368256ddd010d30d98ffc8a0a005f166e9"),
```
Comment (aviks) [inline X/XGBoost/build_tarballs.jl:7]: @ViralBShah where did you get this git-sha from? The previous value is the tagged `0.8.2` version of XGBoost, but I cant find `bf3241` sha in https://github.com/dmlc/xgboost

Comment (giordano) [reply X/XGBoost/build_tarballs.jl:7]: Looks like it's in the `release_0.82` branch: https://github.com/dmlc/xgboost/commit/bf3241368256ddd010d30d98ffc8a0a005f166e9

**Pull request merged**